python/modem-firmware-1.3+/device_credentials_installer.py

```diff
@@ -438,44 +465,58 @@ def main():
         print(local_style('OS detect: Linux={}, MacOS={}, Windows={}'.
                           format(is_linux, is_macos, is_windows)))
 
-    # get a serial port to use
-    port = ask_for_port(args.port, args.all)
-    if port == None:
-        sys.exit(1)
-
-    # let user know which we are using and as what kind of device
-    if args.gateway:
-        is_gateway = True
-    print(local_style('Opening port {} as {}...'.format(port,
-                                                 'gateway' if is_gateway
-                                                 else 'generic device')))
-
-    # try to open the serial port
-    try:
-        ser = serial.Serial(port, 115200, xonxoff= args.xonxoff, rtscts=(not args.rtscts_off),
-                            dsrdtr=args.dsrdtr, timeout=serial_timeout)
-        ser.reset_input_buffer()
-        ser.reset_output_buffer()
-    except serial.serialutil.SerialException:
-        sys.stderr.write(error_style('Port could not be opened; not a device, or open already\n'))
-        sys.exit(2)
-
-    # for gateways, get to the AT command prompt first
-    if is_gateway:
-        print(local_style('Getting to prompt...'))
-        handle_login()
+    if args.rtt:
+        cmd_term_key = 'CRLF'
 
-        print(local_style('Disabling logs...'))
-        write_line('log disable')
-        wait_for_prompt()
+        rtt = rtt_interface.connect_rtt(args.jlink_sn, args.mosh_rtt_hex)
+        if not rtt:
+            sys.stderr.write(error_style('Failed connect to device via RTT'))
+            sys.exit(2)
 
-        print(local_style('Getting to AT mode...'))
-        write_line('at enable')
-        wait_for_prompt(b'to exit AT mode')
+        if not rtt_interface.enable_at_cmds_mosh_rtt(rtt):
```

Comment:
IMO, this call should be built-in to the `rtt_interface.connect_rtt` function

Reply:
i'd prefer to keep it separate.
connecting to a device via RTT and interacting with the MoSH application to switch it to AT cmd mode are very different things.